src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemName.cs

```diff
@@ -140,8 +140,9 @@ private static bool MatchPattern(ReadOnlySpan<char> expression, ReadOnlySpan<cha
             char expressionChar;
 
             Span<int> temp = stackalloc int[0];
-            Span<int> priorMatches = stackalloc int[16];
             Span<int> currentMatches = stackalloc int[16];
+            Span<int> priorMatches = stackalloc int[16];
+            priorMatches.Clear();
```

Comment:
@JeremyKuhne, the code takes a dependency on priorMatches being zero'd out.  Is that intentional?

Reply:
I think it only needs the first element to be set to 0. Not sure if `priorMatches[0] = 0;` would buy much?